src/chainstate/stacks/db/blocks.rs

```diff
@@ -347,41 +350,83 @@ impl StagingMicroblock {
 impl BlockStreamData {
     pub fn new_block(index_block_hash: StacksBlockId) -> BlockStreamData {
         BlockStreamData {
-            block_hash: index_block_hash,
+            index_block_hash: index_block_hash,
             rowid: None,
             offset: 0,
             total_bytes: 0,
+
             is_microblock: false,
+            microblock_hash: BlockHeaderHash([0u8; 32]),
```

Comment:
Should we introduce a `MicroblockTailHash`?

Reply:
Maybe in a separate PR.  This would be a pretty disruptive change, since it would significantly affect the chainstate and the networking code.